<!doctype html>
<html class="no-js" lang="en" data-content_root="./">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="genindex.html" /><link rel="search" title="Search" href="search.html" /><link rel="next" title="celldex" href="api/modules.html" /><link rel="prev" title="celldex" href="index.html" />

    <!-- Generated with Sphinx 7.3.7 and Furo 2024.05.06 -->
        <title>celldex - reference cell type datasets - celldex 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo.css?v=387cc868" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo-extensions.css?v=36a5483c" />
    
    


<style>
  body {
    --color-code-background: #eeffcc;
  --color-code-foreground: black;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" /
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="index.html"><div class="brand">celldex 0.1.0 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="index.html">
  
  
  <span class="sidebar-brand-text">celldex 0.1.0 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <ul class="current">
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Overview</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="api/modules.html">Module Reference</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of Module Reference</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="api/celldex.html">celldex package</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributions &amp; Help</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Authors</a></li>
<li class="toctree-l1"><a class="reference internal" href="changelog.html">Changelog</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="view-this-page">
  <a class="muted-link" href="_sources/readme.md.txt" title="View this page">
    <svg><use href="#svg-eye"></use></svg>
    <span class="visually-hidden">View this page</span>
  </a>
</div>
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <!-- These are examples of badges you might want to add to your README:
     please update the URLs accordingly

[![Built Status](https://api.cirrus-ci.com/github/<USER>/celldex.svg?branch=main)](https://cirrus-ci.com/github/<USER>/celldex)
[![ReadTheDocs](https://readthedocs.org/projects/celldex/badge/?version=latest)](https://celldex.readthedocs.io/en/stable/)
[![Coveralls](https://img.shields.io/coveralls/github/<USER>/celldex/main.svg)](https://coveralls.io/r/<USER>/celldex)
[![Conda-Forge](https://img.shields.io/conda/vn/conda-forge/celldex.svg)](https://anaconda.org/conda-forge/celldex)
[![Monthly Downloads](https://pepy.tech/badge/celldex/month)](https://pepy.tech/project/celldex)
[![Twitter](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Twitter)](https://twitter.com/celldex)
-->
<p><a class="reference external" href="https://pyscaffold.org/"><img alt="Project generated with PyScaffold" src="https://img.shields.io/badge/-PyScaffold-005CA0?logo=pyscaffold" /></a>
<a class="reference external" href="https://pypi.org/project/celldex/"><img alt="PyPI-Server" src="https://img.shields.io/pypi/v/celldex.svg" /></a></p>
<section class="tex2jax_ignore mathjax_ignore" id="celldex-reference-cell-type-datasets">
<h1>celldex - reference cell type datasets<a class="headerlink" href="#celldex-reference-cell-type-datasets" title="Link to this heading">¶</a></h1>
<p>This package provides reference datasets with annotated cell types for convenient use by <a class="reference external" href="https://github.com/biocpy">BiocPy</a> packages and workflows in Python.
These references were sourced and uploaded by the <a class="reference external" href="https://bioconductor.org/packages/devel/data/experiment/html/celldex.html">celldex R/Bioconductor</a> package.</p>
<p>Each dataset is loaded as a <a class="reference external" href="https://bioconductor.org/packages/SummarizedExperiment"><code class="docutils literal notranslate"><span class="pre">SummarizedExperiment</span></code></a> that is ready for further analysis, and may be used for downstream analysis, e.g in the <a class="reference external" href="https://github.com/BiocPy/singler">SingleR Python implementation</a>.</p>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Link to this heading">¶</a></h2>
<p>To get started, install the package from <a class="reference external" href="https://pypi.org/project/celldex/">PyPI</a>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pip<span class="w"> </span>install<span class="w"> </span>celldex
</pre></div>
</div>
</section>
<section id="find-reference-datasets">
<h2>Find reference datasets<a class="headerlink" href="#find-reference-datasets" title="Link to this heading">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">list_references()</span></code> function will display all available reference datasets along with their metadata.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">celldex</span> <span class="kn">import</span> <span class="n">list_references</span>

<span class="n">refs</span> <span class="o">=</span> <span class="n">list_references</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="n">refs</span><span class="p">[[</span><span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="s2">&quot;version&quot;</span><span class="p">]]</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>

<span class="c1">## output</span>
<span class="c1"># |    | name             | version    |</span>
<span class="c1"># |---:|:-----------------|:-----------|</span>
<span class="c1"># |  0 | immgen           | 2024-02-26 |</span>
<span class="c1"># |  1 | blueprint_encode | 2024-02-26 |</span>
<span class="c1"># |  2 | dice             | 2024-02-26 |</span>
</pre></div>
</div>
</section>
<section id="fetch-reference-datasets">
<h2>Fetch reference datasets<a class="headerlink" href="#fetch-reference-datasets" title="Link to this heading">¶</a></h2>
<p>Fetch a dataset as a <a class="reference external" href="https://github.com/biocpy/summarizedexperiment">SummarizedExperiment</a>:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">ref</span> <span class="o">=</span> <span class="n">fetch_reference</span><span class="p">(</span><span class="s2">&quot;immgen&quot;</span><span class="p">,</span> <span class="n">version</span><span class="o">=</span><span class="s2">&quot;2024-02-26&quot;</span><span class="p">)</span>
<span class="n">ref2</span> <span class="o">=</span> <span class="n">fetch_reference</span><span class="p">(</span><span class="s2">&quot;hpca&quot;</span><span class="p">,</span> <span class="s2">&quot;2024-02-26&quot;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="n">ref</span><span class="p">)</span>

<span class="c1">## output</span>
<span class="c1"># class: SummarizedExperiment</span>
<span class="c1"># dimensions: (22134, 830)</span>
<span class="c1"># assays(1): [&#39;logcounts&#39;]</span>
<span class="c1"># row_data columns(0): []</span>
<span class="c1"># row_names(22134): [&#39;Zglp1&#39;, &#39;Vmn2r65&#39;, &#39;Gm10024&#39;, ..., &#39;Ifi44&#39;, &#39;Tiparp&#39;, &#39;Kdm1a&#39;]</span>
<span class="c1"># column_data columns(3): [&#39;label.main&#39;, &#39;label.fine&#39;, &#39;label.ont&#39;]</span>
<span class="c1"># column_names(830): [&#39;GSM1136119_EA07068_260297_MOGENE-1_0-ST-V1_MF.11C-11B+.LU_1.CEL&#39;, &#39;GSM1136120_EA07068_260298_MOGENE-1_0-ST-V1_MF.11C-11B+.LU_2.CEL&#39;, &#39;GSM1136121_EA07068_260299_MOGENE-1_0-ST-V1_MF.11C-11B+.LU_3.CEL&#39;, ..., &#39;GSM920653_EA07068_201207_MOGENE-1_0-ST-V1_TGD.VG4+24AHI.E17.TH_3.CEL&#39;, &#39;GSM920654_EA07068_201214_MOGENE-1_0-ST-V1_TGD.VG4+24ALO.E17.TH_1.CEL&#39;, &#39;GSM920655_EA07068_201215_MOGENE-1_0-ST-V1_TGD.VG4+24ALO.E17.TH_2.CEL&#39;]</span>
<span class="c1"># metadata(0):</span>
</pre></div>
</div>
</section>
<section id="search-for-references">
<h2>Search for references<a class="headerlink" href="#search-for-references" title="Link to this heading">¶</a></h2>
<p>There’s limited number of references right now, but if you want to search for references,</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">res</span> <span class="o">=</span> <span class="n">search_references</span><span class="p">(</span><span class="s2">&quot;human&quot;</span><span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">search_references</span><span class="p">(</span><span class="n">define_text_query</span><span class="p">(</span><span class="s2">&quot;Immun%&quot;</span><span class="p">,</span> <span class="n">partial</span><span class="o">=</span><span class="s2">&quot;True&quot;</span><span class="p">))</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">search_references</span><span class="p">(</span><span class="n">define_text_query</span><span class="p">(</span><span class="s2">&quot;10090&quot;</span><span class="p">,</span> <span class="n">field</span><span class="o">=</span><span class="s2">&quot;taxonomy_id&quot;</span><span class="p">))</span>
</pre></div>
</div>
</section>
<section id="adding-new-reference-datasets">
<h2>Adding new reference datasets<a class="headerlink" href="#adding-new-reference-datasets" title="Link to this heading">¶</a></h2>
<p>These instructions follow the same steps outlined in the <a class="reference external" href="https://github.com/biocpy/scrnaseq">scrnaseq package</a>.</p>
<ol class="arabic">
<li><p>Format your dataset as a <code class="docutils literal notranslate"><span class="pre">SummarizedExperiment</span></code>. Let’s mock a reference dataset:</p>
<p><em><strong>Note: Experiment object must include an assay (‘logcounts’) matrix containing log-normalized counts.</strong></em></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">summarizedexperiment</span> <span class="kn">import</span> <span class="n">SummarizedExperiment</span>
<span class="kn">from</span> <span class="nn">biocframe</span> <span class="kn">import</span> <span class="n">BiocFrame</span>

<span class="n">mat</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">exponential</span><span class="p">(</span><span class="mf">1.3</span><span class="p">,</span> <span class="p">(</span><span class="mi">100</span><span class="p">,</span> <span class="mi">10</span><span class="p">))</span>
<span class="n">row_names</span> <span class="o">=</span> <span class="p">[</span><span class="sa">f</span><span class="s2">&quot;GENE_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">mat</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])]</span>
<span class="n">col_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="s2">&quot;ABCDEFGHIJ&quot;</span><span class="p">)</span>
<span class="n">sce</span> <span class="o">=</span> <span class="n">SummarizedExperiment</span><span class="p">(</span>
     <span class="n">assays</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;logcounts&quot;</span><span class="p">:</span> <span class="n">mat</span><span class="p">},</span>
     <span class="n">row_data</span><span class="o">=</span><span class="n">BiocFrame</span><span class="p">(</span><span class="n">row_names</span><span class="o">=</span><span class="n">row_names</span><span class="p">),</span>
     <span class="n">column_data</span><span class="o">=</span><span class="n">BiocFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;label.fine&quot;</span><span class="p">:</span> <span class="n">col_names</span><span class="p">}),</span>
<span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Assemble the metadata for your reference dataset. This should be a dictionary as specified in the <a class="reference external" href="https://github.com/ArtifactDB/bioconductor-metadata-index">Bioconductor metadata schema</a>. Check out some examples from <code class="docutils literal notranslate"><span class="pre">fetch_metadata()</span></code>. Note that the <code class="docutils literal notranslate"><span class="pre">application.takane</span></code> property will be automatically added later, and so can be omitted from the list that you create.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">meta</span> <span class="o">=</span> <span class="p">{</span>
     <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;New reference dataset&quot;</span><span class="p">,</span>
     <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="s2">&quot;This is a new reference dataset&quot;</span><span class="p">,</span>
     <span class="s2">&quot;taxonomy_id&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;10090&quot;</span><span class="p">],</span>  <span class="c1"># NCBI ID</span>
     <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;GRCm38&quot;</span><span class="p">],</span>  <span class="c1"># genome build</span>
     <span class="s2">&quot;sources&quot;</span><span class="p">:</span> <span class="p">[{</span><span class="s2">&quot;provider&quot;</span><span class="p">:</span> <span class="s2">&quot;GEO&quot;</span><span class="p">,</span> <span class="s2">&quot;id&quot;</span><span class="p">:</span> <span class="s2">&quot;GSE12345&quot;</span><span class="p">}],</span>
     <span class="s2">&quot;maintainer_name&quot;</span><span class="p">:</span> <span class="s2">&quot;Jayaram kancherla&quot;</span><span class="p">,</span>
     <span class="s2">&quot;maintainer_email&quot;</span><span class="p">:</span> <span class="s2">&quot;jayaram.kancherla@gmail.com&quot;</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</li>
<li><p>Save your <code class="docutils literal notranslate"><span class="pre">SummarizedExperiment</span></code>  object to disk with <code class="docutils literal notranslate"><span class="pre">save_reference()</span></code>. This saves the reference dataset into a “staging directory” using language-agnostic file formats - check out the <a class="reference external" href="https://github.com/artifactdb">ArtifactDB</a> framework for more details.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">tempfile</span>
<span class="kn">from</span> <span class="nn">celldex</span> <span class="kn">import</span> <span class="n">save_reference</span>

<span class="c1"># replace tmp with a staging directory</span>
<span class="n">staging_dir</span> <span class="o">=</span> <span class="n">tempfile</span><span class="o">.</span><span class="n">mkdtemp</span><span class="p">()</span>
<span class="n">save_reference</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span> <span class="n">staging_dir</span><span class="p">,</span> <span class="n">meta</span><span class="p">)</span>
</pre></div>
</div>
<p>You can check that everything was correctly saved by reloading the on-disk data for inspection:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">dolomite_base</span> <span class="k">as</span> <span class="nn">dl</span>

<span class="n">dl</span><span class="o">.</span><span class="n">read_object</span><span class="p">(</span><span class="n">staging_dir</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Wait for us to grant temporary upload permissions to your GitHub account.</p></li>
<li><p>Upload your staging directory to <a class="reference external" href="https://github.com/ArtifactDB/gypsum-worker"><strong>gypsum</strong> backend</a> with <code class="docutils literal notranslate"><span class="pre">upload_reference()</span></code>. On the first call to this function, it will automatically prompt you to log into GitHub so that the backend can authenticate you. If you are on a system without browser access (e.g., most computing clusters), a <a class="reference external" href="https://github.com/settings/tokens">token</a> can be manually supplied via <code class="docutils literal notranslate"><span class="pre">set_access_token()</span></code>.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">celldex</span> <span class="kn">import</span> <span class="n">upload_reference</span>

<span class="n">upload_reference</span><span class="p">(</span><span class="n">staging_dir</span><span class="p">,</span> <span class="s2">&quot;my_dataset_name&quot;</span><span class="p">,</span> <span class="s2">&quot;my_version&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>You can check that everything was successfully uploaded by calling <code class="docutils literal notranslate"><span class="pre">fetch_reference()</span></code> with the same name and version:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">celldex</span> <span class="kn">import</span> <span class="n">fetch_reference</span>

<span class="n">fetch_reference</span><span class="p">(</span><span class="s2">&quot;my_dataset_name&quot;</span><span class="p">,</span> <span class="s2">&quot;my_version&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>If you realized you made a mistake, no worries. Use the following call to clear the erroneous dataset, and try again:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">gypsum_client</span> <span class="kn">import</span> <span class="n">reject_probation</span>

<span class="n">reject_probation</span><span class="p">(</span><span class="s2">&quot;celldex&quot;</span><span class="p">,</span> <span class="s2">&quot;my_dataset_name&quot;</span><span class="p">,</span> <span class="s2">&quot;my_version&quot;</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Comment on the PR to notify us that the dataset has finished uploading and you’re happy with it. We’ll review it and make sure everything’s in order. If some fixes are required, we’ll just clear the dataset so that you can upload a new version with the necessary changes. Otherwise, we’ll approve the dataset. Note that once a version of a dataset is approved, no further changes can be made to that version; you’ll have to upload a new version if you want to modify something.</p></li>
</ol>
<!-- pyscaffold-notes -->
</section>
<section id="note">
<h2>Note<a class="headerlink" href="#note" title="Link to this heading">¶</a></h2>
<p>This project has been set up using PyScaffold 4.5. For details and usage
information on PyScaffold see <a class="reference external" href="https://pyscaffold.org/">https://pyscaffold.org/</a>.</p>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="api/modules.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">celldex</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="index.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Home</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Jayaram Kancherla
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">celldex - reference cell type datasets</a><ul>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#find-reference-datasets">Find reference datasets</a></li>
<li><a class="reference internal" href="#fetch-reference-datasets">Fetch reference datasets</a></li>
<li><a class="reference internal" href="#search-for-references">Search for references</a></li>
<li><a class="reference internal" href="#adding-new-reference-datasets">Adding new reference datasets</a></li>
<li><a class="reference internal" href="#note">Note</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="_static/documentation_options.js?v=01f34227"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/scripts/furo.js?v=4e2eecee"></script>
    </body>
</html>